### PR TITLE
characteristic -> automat; remove attrs name conflict in tests

### PR DESCRIPTION
**Pull request by markrwilliams**
What it says on the tin!

### CODE REVIEW

File: automat/_methodical.py [hunk 1]
```diff
@@ -68,14 +70,16 @@ def _transitionerFromInstance(oself, symbol, automaton):
     return transitioner
 
 
-@with_repr(['method'])
-@attributes(['automaton', 'method', 'symbol',
-             Attribute('collectors', default_factory=dict)],
-            apply_with_cmp=False)
+@attr.s(cmp=False, hash=False)
```
Comment (markrwilliams) [inline automat/_methodical.py]: `characteristic`'s `with_cmp` implied `__hash__`, but `attrs` decouples them.  I included `cmp=False` to keep the API the same.

**Comment by markrwilliams** [comment]
Ok so I'm just going to leave the goofy typo in the title, though through Git trickery™ I erased it from the commit.

**Comment by hynek** [comment]
My name is Hynek and I approve of these changes.

**Comment by glyph** [comment]
L G T M.

**Pull request merged**